### PR TITLE
FLUID-5606: Updated to be a relay component

### PR DESCRIPTION
**Pull request by jobara**
Updated the component to be a viewRelay compnonent.
http://issues.fluidproject.org/browse/FLUID-5606

There were several TODO's scattered throughout the code that haven't yet been addressed. Those should be taken care of as part of http://issues.fluidproject.org/browse/FLUID-5607

### CODE REVIEW

**Comment by michelled** [comment]
@jobara , can you please update the TODO comments to mention FLUID-5607 and update FLUID-5607 with the specific changes that need to happen?

**Comment by jobara** [comment]
@michelled a couple of those TODO's are a bit vague and I'm not exactly sure what needs to be done of them. Maybe we should talk to @amb26 about what exactly needs to be done.

**Comment by jobara** [comment]
@michelled I talked to @amb26 this morning and made a couple more changes to the pull request. I've also updated the description of [FLUID-5607](http://issues.fluidproject.org/browse/FLUID-5606) which discusses the remaining work based on the TODO's in the code.

File: src/components/tooltip/js/Tooltip.js [hunk 1]
```diff
@@ -199,8 +188,11 @@ var fluid_2_0 = fluid_2_0 || {};
             onDestroy: "fluid.tooltip.doDestroy"
```
Comment (cindyli) [inline src/components/tooltip/js/Tooltip.js]: Can we take the chance to namespace these event listeners? Thanks.

**Comment by jobara** [comment]
@cindyli ready for another round of review

**Comment by cindyli** [comment]
Merged @ a930d31ca6430dad67566e712ccad5d333d74ddb

**Pull request merged**